### PR TITLE
Project changes

### PR DESCRIPTION
**Pull request by Eitot**
Some additional housekeeping.

- Moves some source files to more appropriate places, e.g. the NSURL+Utils files are external/third-party to Vienna.
- Sets some default project settings for new files, specifically spaces instead of tabs and the default prefix “VNA”. This might make it easier for contributors to use the right settings when they use Xcode.
- Modifies SwiftLint to suppress some warnings, per the discussion on Slack.

### CODE REVIEW

**Review by barijaona — COMMENTED** [review]
It's beautiful, but this raises comments similar to #1362 : Is it possible to wait until after #1312 is finished? There will be a lot more work for @TAKeanice to do if we change AppController and friends this much now.

File: Vienna/Sources/Application/AppController.m [hunk 1]
```diff
@@ -552,15 +562,17 @@ -(BOOL)application:(NSApplication *)theApplication openFile:(NSString *)filename
...
+            [self.mainWindow makeKeyAndOrderFront:self];
+        }
         if (url != nil && !db.readOnly)
         {
             [self.rssFeed newSubscription:self.mainWindow underParent:self.foldersTree.groupParentSelection initialURL:url.absoluteString];
 		    return YES;
         }
-        else
-        	return NO;
+        else {
```
[the remark this answers is not in the document]
Comment (Eitot) [reply Vienna/Sources/Application/AppController.m:573]: Not without doing more comprehensive formatting, unfortunately. My aim was to just add braces without making too many changes to the surrounding formatting. As soon as the WebKit changes are merged, we should pass a formatter over the entire project once and be done with it.

[the remark this answers is not in the document]
Comment (Eitot) [reply Vienna/Sources/Application/AppController.m]: I removed the braces commit for now.

**Comment by Eitot** [comment]
I removed the braces commit. Let’s do that another time.

**Comment by TAKeanice** [comment]
Actually I don´t see any problems here, as there are not many actual changes to any files. The moved files don´t touch any of my work, and the Swift Linter warning changes actually help because it found many issues in the new files in the webkit_browser branch.

**Review by TAKeanice — APPROVED** [review]

**Comment by Eitot** [comment]
@TAKeanice Any other SwiftLint warnings you find too annoying?

**Comment by TAKeanice** [comment]
Well I am not sure how to approach the unavailable function warning. For a function on its own it may be useful, but for "abstract" functions (which don't exist in swift) I should be able to say `fatalError("must be overridden")` without warning.

**Comment by Eitot** [comment]
What does an abstract function do? SwiftLint just recommends adding the `@unavailable` attribute to prevent you from accidentally calling the function elsewhere.

**Comment by Eitot** [comment]
I also updated the .gitignore file to get rid of some ancient cruft. Github happens to have an up-to-date repository that can be used as a reference.

File: .gitignore [hunk 2]
```diff
@@ -1,178 +1,22 @@
...
-#
-# NB: slash on the end, so we only remove the FOLDER, not any files that were badly named "DerivedData"
-
-DerivedData/
-
-# NB: slash on the end, so we only remove the FOLDER, not any files that were badly named "build"
-
+# Xcode
 build/
-Build/
```
Comment (barijaona) [inline .gitignore:64]: Need to keep Build/ (cf. Xcode's  Project Settings…)

**Review by barijaona — APPROVED** [review]

**Pull request merged**